concent_api/verifier/tasks.py

```diff
@@ -106,264 +107,252 @@ def blender_verification_order(
-                ErrorCode.VERIFIER_FILE_DOWNLOAD_FAILED.name
-            )
-            return
-        except Exception as exception:
-            log_string_message(
-                logger,
-                f'blender_verification_order for SUBTASK_ID {subtask_id} failed with error {exception}.'
-                f'ErrorCode: {ErrorCode.VERIFIER_FILE_DOWNLOAD_FAILED.name}'
-            )
-            verification_result.delay(
-                subtask_id,
-                VerificationResult.ERROR.name,
-                str(exception),
-                ErrorCode.VERIFIER_FILE_DOWNLOAD_FAILED.name
-            )
-            raise
+    download_archives_from_storage(
+        file_transfer_token,
+        subtask_id,
+        package_paths_to_downloaded_archive_names
+    )
 
-    # If any file which is supposed to be unpacked from archives already exists, finish with error and raise exception.
-    for package_file_path in (source_package_path, result_package_path):
-        package_files_list = get_files_list_from_archive(
-            generate_verifier_storage_file_path(package_file_path)
-        )
-        if list(set(os.listdir(settings.VERIFIER_STORAGE_PATH)).intersection(package_files_list)):
-            verification_result.delay(
-                subtask_id,
-                VerificationResult.ERROR.name,
-                f'One of the files which are supposed to be unpacked from {package_file_path} already exists.',
-                ErrorCode.VERIFIER_UNPACKING_ARCHIVE_FAILED.name
-            )
-            raise VerificationError(  # TODO: write a test for this case
-                f'One of the files which are supposed to be unpacked from {package_file_path} already exists.',
-                ErrorCode.VERIFIER_UNPACKING_ARCHIVE_FAILED,
-            )
+    validate_downloaded_archives(subtask_id, package_paths_to_downloaded_archive_names.values())
 
-    # Verifier unpacks the archive with project source.
-    for file_path, download_file_name in package_paths_to_downloaded_file_names.items():
-        try:
-            unpack_archive(
-                os.path.basename(file_path)
-            )
-        except BadZipFile as exception:
-            log_string_message(
-                logger,
-                f'Verifier failed to unpack the archive with project source with error {exception} '
-                f'SUBTASK_ID {subtask_id}. '
-                f'ErrorCode: {ErrorCode.VERIFIER_UNPACKING_ARCHIVE_FAILED.name}'
-            )
-            verification_result.delay(
-                subtask_id,
-                VerificationResult.ERROR.name,
-                str(exception),
-                ErrorCode.VERIFIER_UNPACKING_ARCHIVE_FAILED.name
-            )
-            return
+    unpack_archives(package_paths_to_downloaded_archive_names.values(), subtask_id)
 
-    # Verifier runs blender process.
-    try:
-        completed_process = run_blender(
-            scene_file,
-            output_format,
+    frame_number = 1
```

Comment:
I know this is temporary but you can already add it as a list and render in loop.

Reply:
I would prefer not to do this in this PR -> it has a lot of changes as it is ;)
I can do it in separate issue